services/motion/builtin/move_request.go

```diff
@@ -837,3 +838,25 @@ func toGeoPosePlanSteps(posesByComponent []motionplan.PlanStep, geoPoses []spati
 	}
 	return steps, nil
 }
+
+// This function will check the orientation of the startPose of a 2d move request, and ensure that it is normal to the XY plane. If it is
+// not, it will be altered such that it is (accounting for e.g. an ourdoor rover with one wheel on a rock). If the orientation is such that
+// the robot is pointed directly up or down (or is upside-down), an error is returned.
+func correctStartPose(startPose spatialmath.Pose) (spatialmath.Pose, error) {
```

Comment:
I can't say I fully understand this math so I'm going to rely on @raybjork to confirm the math is correct.

Reply:
@nicksanford I added some comments to make it more clear what the math is doing and why it works; could you let me know if this helps?

Reply:
These comments were very helpful & cleared up a lot. Thanks for adding them!